src/jit/lir.cpp

```diff
@@ -972,11 +972,19 @@ void LIR::Range::InsertAtEnd(Range&& range)
 // Arguments:
 //    node - The node to remove. Must be part of this range.
 //
-void LIR::Range::Remove(GenTree* node)
+void LIR::Range::Remove(GenTree* node, bool markOperandsUnused)
 {
     assert(node != nullptr);
     assert(Contains(node));
 
+    if (markOperandsUnused)
+    {
+        for (GenTree* operand : node->Operands())
+        {
+            operand->gtLIRFlags |= Flags::IsUnusedValue;
+        }
+    }
```

Comment:
Maybe you could add this as a small GenTree helper to avoid duplicating this both here and in liveness: `node->MarkOperandsUnused()`. Yeah, it's small; your choice.